src/gc/gc.cpp

```diff
@@ -1657,12 +1682,12 @@ static void leave_spin_lock(GCSpinLock *pSpinLock)
 //the gc thread call WaitLonger.
 void WaitLonger (int i
 #ifdef SYNCHRONIZATION_STATS
-    , VOLATILE(GCSpinLock)* spin_lock
+    , GCSpinLock* spin_lock
 #endif //SYNCHRONIZATION_STATS
     )
 {
 #ifdef SYNCHRONIZATION_STATS
-    (spin_lock->num_wait_longer)++;
+    Interlocked::Increment(&spin_lock->num_wait_longer);
```

Comment:
I am on the fence about these ++ -> Interlocked add changes. I didn't want to use Interlocked 'cause it's expensive and these counts don't need to be perfectly accurate. Have you measured the overhead with these Interlocked operations when you define the STATS? If you take a test that just does straight up allocations, how much overhead do you notice with these Interlocks?